src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/v0/MigrationIndexer.scala

```diff
@@ -0,0 +1,125 @@
+
+  // $COVERAGE-OFF$
+  private def run(): ActorRef = KafkaConsumer.start[v0.Event](consumerSettings, index, topic, s"$topic-migration")
+
+  private def index(event: v0.Event): Future[Unit] = event.id match {
+    case pattern() =>
+      process(event).value.map {
+        case Right(res) => log.debug(s"Migrated v0 event '${event.id}' -> resource '${res.id}'")
+        case Left(e)    => log.error(s"Couldn't migrate v0 event '${event.id}'", e)
+      }.runAsync
+    case _ => Future.unit
+  }
+  // $COVERAGE-ON$
+
+  private[v0] def process(event: v0.Event): EitherT[Task, Rejection, Resource] = {
+    implicit val author: Identity = event.meta.toIdentity
+    event match {
+      case InstanceCreated(id, _, meta, value) =>
+        repo.create(toId(id), extractSchema(id), extractTypes(value), value, meta.instant)
+      case ContextCreated(id, _, meta, value) =>
+        repo.create(toId(id), resourceSchema, resourceTypes, value, meta.instant)
+      case SchemaCreated(id, _, meta, value) =>
+        repo.create(toId(id), resourceSchema, schemaTypes, value, meta.instant)
+
+      case InstanceUpdated(id, rev, meta, value) => repo.update(toId(id), rev, extractTypes(value), value, meta.instant)
+      case ContextUpdated(id, rev, meta, value)  => repo.update(toId(id), rev, resourceTypes, value, meta.instant)
+      case SchemaUpdated(id, rev, meta, value)   => repo.update(toId(id), rev, schemaTypes, value, meta.instant)
+
+      case InstanceDeprecated(id, rev, meta) => repo.deprecate(toId(id), rev, meta.instant)
+      case ContextDeprecated(id, rev, meta)  => repo.deprecate(toId(id), rev, meta.instant)
+      case SchemaDeprecated(id, rev, meta)   => repo.deprecate(toId(id), rev, meta.instant)
+
+      case InstanceAttachmentCreated(id, rev, meta, value) =>
+        repo.unsafeAttach(toId(id), rev, value.toBinaryAttributes, meta.instant)
+      case InstanceAttachmentRemoved(id, rev, meta) =>
+        repo
+          .get(toId(id))
+          .toRight(NotFound(toId(event.id).ref))
+          .flatMap { res =>
+            res.attachments.toList match { // v0 instances can have only one attachment
+              case head :: Nil => repo.unattach(res.id, rev, head.filename, meta.instant)
+              case _           => EitherT.fromEither[Task](Left(UnexpectedState(res.id.ref)))
+            }
+          }
+
+      case _: ContextPublished | _: SchemaPublished =>
+        repo
+          .get(toId(event.id))
+          .toRight(NotFound(toId(event.id).ref))
+          .flatMap(res => repo.update(res.id, event.rev, res.types, res.value, event.meta.instant))
+    }
+  }
+
+  private def toId(id: String): Id[ProjectRef] = Id(projectRef, base + id)
+
+  private def extractTypes(value: Json): Set[AbsoluteIri] = {
+    val types = value.asObject
+      .flatMap(_.apply("@type"))
+      .map(tpe => tpe.asString.toList ++ tpe.asArray.toList.flatMap(_.flatMap(_.asString))) match {
+      case Some(tpes) => tpes.flatMap(Iri.absolute(_).toSeq)
```

Comment:
Is not enough to convert to Iri.absolute. Things like `a:b` can be converted to an absolute iri surprisingly. You should do something like:
```scala
tpes.flatMap(Iri.url(_).toSeq.filter(_.scheme.value.matches("http(s?)")))
```